mixer/pkg/adapter/requestData_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapter
+
+import (
+	"context"
+	"reflect"
+	"testing"
+)
+
+func TestRequestDataFromContext(t *testing.T) {
```

Comment:
I feel like we should have a test where we load the context with other keys (maybe `const badKey int = 0`) to validate key collision avoidance (as we call it our in our comments).

Reply:
ha. this is because the github UI swallowed (or so I thought) my original comment. I guess it ultimately sent it anyways. sorry for the confusion.